docs/multiplatform.md

```diff
@@ -0,0 +1,89 @@
+# Multiplatform
+
+This document covers using kotlin-inject in a multiplatform project.
+
+## Download
+
+Using [ksp](https://github.com/google/ksp)
+
+`settings.gradle`
+
+```kotlin
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        mavenCentral()
+    }
+}
+
+dependencyResolutionManagement {
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+```
+
+`build.gradle`
+
+Add Plugin
+```kotlin
+plugins {
+    id("org.jetbrains.kotlin.multiplatform") version "1.9.21"
+    id("com.android.library")
+    id("com.google.devtools.ksp") version "1.9.21-1.0.16"
+}
+```
+
+Add Runtime dependency in commonMain
+```kotlin
+sourceSets {
+    androidMain.dependencies {
+        //...
+    }
+    commonMain.dependencies {
+        implementation("me.tatarka.inject:kotlin-inject-runtime:0.6.3")
+    }
+}
+```
+If you're targetting iOS, you need to add each of the chipset as separate sourcesets to include generated files.
+Same goes to any other LLVM (MacOS, TVOS etc.) you're targetting.
+```kotlin
+val iosX64Main by getting {
+    dependencies {
+        dependsOn(iosMain.get())
+        kotlin.srcDir("build/generated/ksp/iosX64/iosX64Main/kotlin")
```

Comment:
Do we still need to add these as srcDirs? our multiplatform sample doesn't seem to have it https://github.com/evant/kotlin-inject-samples/blob/main/multiplatform/greeter/shared/build.gradle.kts, in theory it should be no longer necessary as of [KSP 1.8.0-1.0.9](https://kotlinlang.org/docs/ksp-quickstart.html#make-ide-aware-of-generated-code), but I haven't taken the time to verify.

Reply:
I needed this, compiler wasn't detecting the generated folder without this.
I tested it in 3 different projects (one where I'm actually using it on prod in JioCinema app, another one is in the courses I do on KMP, and lastly in the demo project for this document)

Reply:
I've updated the greeter sample and the IDE is resolving references fine without these additional srcDir declarations. Removing them from the doc but let me know if there's something I'm missing.